examples/mnit_moles/src/moles.nit

```diff
@@ -90,38 +107,114 @@ class Hole
 			ey > y - dy and ey < y
 	end
 
-	fun hit
+	# Draw this hole and content to `display`
+	fun draw(display: Display)
 	do
-		if hitted then return
-
-		if trap then
-			up = false
-			game.points -= 5
-			if game.points < 0 then game.points = 0
-		else
-			hitted = true
-			game.points += 1
+		# The hole itself
+		var img = app.assets.empty
+		var dx = 300.0*display_scale
+		var dy = 256.0*display_scale
+		img.scale = display_scale
+		display.blit(img, x-dx.to_i+display_offset_x, y-dy.to_i+display_offset_y)
+
+		# The mole in the hole, or other content
+		var content = self.content
+		if content != null then
+			content.draw(display, x, y)
 		end
 	end
 end
 
+# Content of a `Hole`
+class HoleContent
+	# Image
+	var img: Image
+
+	# Offset of the horizontal center of the hole
+	var dx: Float
+
+	# Offset of the vertical center of the hole
+	var dy: Float
+
+	# Hit this hole content
+	fun hit(game: Game, hole: Hole, event: PointerEvent) do end
+
+	# Draw this content to `display`
+	fun draw(display: Display, x, y: Int)
+	do
+		img.scale = display_scale
+		display.blit(img,
+			x-dx.to_i+display_offset_x,
+			y-dy.to_i+display_offset_y)
+	end
+end
+
+# A mole in a hole
+class Mole
+	super HoleContent
+
+	# Points value when hit
+	var value: Int
+
+	redef fun hit(game, hole, event)
+	do
+		game.points += value
+		hole.content = game.down
+	end
+end
+
+# A trap held out of a hole
+class Trap
+	super HoleContent
+
+	# Points penalty when hit
+	var penalty: Int
```

Comment:
It it strange that the `penalty` is an attribute assigned by the new instead of a class information (eg `redef fun penalty do return 20` is the big trap class)